cpp/CMakeLists.txt

```diff
@@ -22,12 +22,48 @@ set(CMAKE_CXX_FLAGS_RELWITHDEBINFO
 set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG")
 set(CMAKE_SHARED_LIBRARY_PREFIX "")
 
+# Install testing dependencies
+find_package(Threads REQUIRED)
+
+# Install gtest
```

Comment:
Install dependencies with respect to different flags: RELEASE/DEBUG/TEST

Reply:
What is the purpose of every time building with tests on MAGE? If someone is purely interested in running the MAGE, tests shouldn't be something that is generated once the build is initiated?

Reply:
As the users will build on very different machines, you want to ensure them that they did everything correctly (and that the modules can be correctly built on their machines). Also, you copy the libraries in a different folder so the extra test files shouldn't matter. 
If the user still has some problems with the module, we can be more confident that there is a bug in the implementation of that specific module.